kyaml/yaml/rnode.go

```diff
@@ -248,6 +377,619 @@ func (rn *RNode) IsNilOrEmpty() bool {
+	}
+}
+
+// NestedInt returns a nested int.
+func (o *KubeObject) NestedInt(fields ...string) (int, bool, error) {
+	scalarNode, found, err := o.nestedScalar(fields...)
+	if err != nil || !found {
+		return 0, found, err
+	}
+
+	switch scalarNode.value.Tag {
+	case NodeTagInt:
+		i, err := strconv.Atoi(scalarNode.value.Value)
+		if err != nil {
+			return 0, true, fmt.Errorf("unable to parse %q as int: %w", scalarNode.value.Value, err)
+		}
+		return i, true, nil
+	default:
+		return 0, true, fmt.Errorf("node was not an int, was %v", scalarNode.value.Tag)
+	}
+}
+
+// NestedFloat returns a nested float.
+func (o *KubeObject) NestedFloat(fields ...string) (float64, bool, error) {
+	scalarNode, found, err := o.nestedScalar(fields...)
+	if err != nil || !found {
+		return 0, found, err
+	}
+
+	switch scalarNode.value.Tag {
+	case NodeTagFloat:
+		f, err := strconv.ParseFloat(scalarNode.value.Value, 64)
+		if err != nil {
+			return 0, true, fmt.Errorf("unable to parse %q as float: %w", scalarNode.value.Value, err)
+		}
+		return f, true, nil
+	default:
+		return 0, true, fmt.Errorf("node was not a float, was %v", scalarNode.value.Tag)
+	}
+}
+
+// NestedBool returns a nested boolean.
+func (o *KubeObject) NestedBool(fields ...string) (bool, bool, error) {
+	scalarNode, found, err := o.nestedScalar(fields...)
+	if err != nil || !found {
+		return false, found, err
+	}
+
+	switch scalarNode.value.Tag {
+	case NodeTagBool:
+		b, err := strconv.ParseBool(scalarNode.value.Value)
+		if err != nil {
+			return false, true, fmt.Errorf("unable to parse %q as bool: %w", scalarNode.value.Value, err)
+		}
+		return b, true, nil
+	default:
+		return false, true, fmt.Errorf("node was not a bool, was %v", scalarNode.value.Tag)
+	}
+}
+
+// LineComment return the line comment of the field.
+func (o *KubeObject) LineComment(fields ...string) (string, error) {
+	node, found, err := o.nestedValue(fields...)
+	if err != nil {
+		return "", err
+	}
+	if !found {
+		return "", fmt.Errorf("unable to get get line comment for field %v, since it doesn't exist", fields)
+	}
+	return node.value.LineComment, nil
+}
+
+// LineComment return the head comment of the field.
+func (o *KubeObject) HeadComment(fields ...string) (string, error) {
+	node, found, err := o.nestedValue(fields...)
+	if err != nil {
+		return "", err
+	}
+	if !found {
+		return "", fmt.Errorf("unable to get get line comment for field %v, since it doesn't exist", fields)
+	}
+	return node.value.HeadComment, nil
+}
+
+// SetOrDie sets a nested field located by fields to the value provided as val.
+// It will panic if it encounters any error.
+func (o *KubeObject) SetOrDie(val interface{}, fields ...string) {
+	if err := o.Set(val, fields...); err != nil {
+		panic(err)
+	}
+}
+
+// Set sets a nested field located by fields to the value provided as val. val
+// should not be a yaml.RNode. If you want to deal with yaml.RNode, you should
```

Comment:
I may miss something. In [ln 763-764](https://github.com/kubernetes-sigs/kustomize/pull/4319/files#diff-3e6720332daec570bb6996084aca231f72caa46e83e352fa61b6937209a1eb79R764), the code accepts val as a yaml.RNode.  Should the code be fixed to raise errors if type RNode is detected?

Reply:
Updated